types/lhr/lhr.d.ts

```diff
@@ -56,6 +56,8 @@ interface Result {
   };
   /** An array containing the result of all stack packs. */
   stackPacks?: Result.StackPack[];
+  /** Entity classification for this Lighthouse run. */
+  entityClassification?: Result.EntityClassification;
```

Comment:
Can this actually ever be undefined? I see there's a failure mode when the devtools log is missing... but if that's the case then the entire LHR is moot. Maybe can return an empty default for that case and we can make this always present

Reply:
Older LHRs?

Reply:
In the past we've always defined the LHR type as what LH will create, _today_. And used prepareReportResult to do any necessary back compat.

Although, now that we're about to have actual type support for Node users, maybe we should consider the use case of the LHR type being applied to older formats (in which case, keeping this optional is the right move). @adamraine wdyt?

Another option - we extract all the back-compat stuff from prepareReportResult into a new function exposed as the core api that will "update" a LHR. We can then document that the function should be used when loading LHRs from disk, or elsewhere, and wanting to use the types associated with the current LH version.

None of this is blocking this PR

Reply:
This one's different from say `fullPageScreenshot`, where the previous LHR can be transformed into latest by moving the data field. For older reports, entity classification would be absent. Our code, [`renderReport`](https://github.com/GoogleChrome/lighthouse/blob/ae39bb54debc65e90c39924095698b2865119a2b/report/renderer/report-renderer.js#L49) for example, takes in `LR.Result` type prior to `prepareReportResult`, so this marked as optional/undefined when missing seems apt.

Re: migrations, we could generalize this to be a series of data migration functions that transform LHR from a certain version to the next, and assuming we run them sequentially, would migrate an outdated LHR JSON all the way to latest — very similar to [database migrations](https://knexjs.org/guide/migrations.html#transactions-in-migrations). This would also allow us to delete migrations of versions that are no longer needed, based on how old a version we want to supports vs. carrying all our migrations with each time (as we do today).

Reply:
> Although, now that we're about to have actual type support for Node users, maybe we should consider the use case of the LHR type being applied to older formats (in which case, keeping this optional is the right move). @adamraine wdyt?

I would prefer the lhr types to reflect the current lhr structure and not keep stuff optional for the sake of back-porting.

Reply:
Keeping it non-nullable for snapshot mode!